content/en/opentelemetry/agent/agent_with_custom_components.md

```diff
@@ -105,8 +105,9 @@ Build your custom Datadog Agent image and push it to a container registry.
 1. Build the image with Docker:
    ```shell
    docker build . -t agent-otel --no-cache \
-     --build-arg AGENT_VERSION="{{< version key="agent_tag_jmx" >}}" \
-     --build-arg AGENT_BRANCH="{{< version key="agent_branch" >}}"
+     --build-arg AGENT_VERSION="{{< version key="agent_tag" >}}" \
+     --build-arg AGENT_BRANCH="{{< version key="agent_branch" >}}" \
+     --build-arg AGENT_REPO="datadog/agent" 
```

Comment:
As mentioned, since customers should use the release branch, I would maybe change the docs to indicate at `step 0` to switch to the corresponding release branch/tag. We can use the `agent_version` or `agent_branch` variables for this, I think `agent_version` is better.

I would then remove the `AGENT_REPO` build arg, because it would be properly set in the Dockerfile, and should be transparent to the customer. I'd like to keep `AGENT_REPO` as a bit of a hidden feature.

Reply:
In this [PR](https://github.com/DataDog/datadog-agent/pull/31490), `AGENT_REPO=datadog/agent-dev`. We have to set `AGENT_REPO` to `datadog/agent` when using tag, like `7.65.0` because tags are defined in repo  `datadog/agent` . Did I miss something?